src/include/encoding.h

```diff
@@ -129,6 +125,28 @@ WRITE_INTTYPE_ENCODER(int32_t, le32)
 WRITE_INTTYPE_ENCODER(uint16_t, le16)
 WRITE_INTTYPE_ENCODER(int16_t, le16)
 
+// -----------------------------------
+// float types
+// FIXME: this only performs endian-conversion, but assumes that the
+// floating-point formats are otherwise the same
+
+#define WRITE_FLTTYPE_ENCODER(type, itype, etype)			\
+  inline void encode(type v, ::ceph::bufferlist& bl, uint64_t features=0) { \
+    ceph_assert(sizeof(type) == sizeof(itype));				\
+    ceph_##etype e;					                \
+    e = *reinterpret_cast<itype *>(&v);                                 \
+    ::ceph::encode_raw(e, bl);						\
```

Comment:
```suggestion
#ifdef CEPH_BIG_ENDIAN
    ceph_##etype e;					                \
    e = *reinterpret_cast<itype *>(&v);                                 \
    ::ceph::encode_raw(e, bl);						\
#else
    ::ceph::encode_raw(v, bl);
#endif
```

Reply:
I hadn't done that initially because the additional conversions should be optimized away by the compiler on a little-endian platform.  But I guess we can do it with an #ifdef if you prefer.  It'll need to be placed outside the #define, however.  Let me know if you like the updated implementation, or if you'd prefer me to take out the #ifdef again.

Reply:
if you are sure that the version on little endian is optimized away, surely we should drop `#ifdef` optimizations.

Reply:
I've had a quick look on x86 and it does appear to be optimized away as it should.  Removed the #ifdef.